File: android/src/main/java/com/margelo/nitro/camera/facedetector/HybridImageFaceDetectorFactory.kt
package com.margelo.nitro.camera.facedetector

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
@Keep
class HybridImageFaceDetectorFactory : HybridImageFaceDetectorFactorySpec() {
  @DoNotStrip
  @Keep
  override fun createImageFaceDetector(options: FaceDetectorOptions): HybridImageFaceDetectorSpec {
    return HybridImageFaceDetector(options)
  }
}

File: ios/HybridImageFaceDetectorFactory.swift
import NitroModules
import VisionCamera

class HybridImageFaceDetectorFactory: HybridImageFaceDetectorFactorySpec {
  func createImageFaceDetector(options: ImageFaceDetectorOptions) throws -> any HybridImageFaceDetectorSpec {
    return HybridImageFaceDetector(options)
  }
}
